Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertRebar.cs

```diff
@@ -1,175 +1,332 @@
+      var speckleStartHook = speckleRebar.startHook as RevitRebarHook;
+      RebarHookType startHook = null;
+      RebarHookOrientation startHookOrientation = RebarHookOrientation.Right;
+      if (speckleStartHook != null)
+      {
+        startHook = RebarHookToNative(speckleStartHook);
+        Enum.TryParse(speckleStartHook.orientation, out startHookOrientation);
+      }
+      var speckleEndHook = speckleRebar.endHook as RevitRebarHook;
+      RebarHookType endHook = null;
+      RebarHookOrientation endHookOrientation = RebarHookOrientation.Right;
+      if (speckleEndHook != null)
+      {
+        endHook = RebarHookToNative(speckleEndHook);
+        Enum.TryParse(speckleEndHook.orientation, out endHookOrientation);
+      }
 
-      // get host element
-      var host = GetExistingElementByApplicationId(speckleRevitRebar.host);
-      if (host == null)
+      // get the shape curves
+      List<DB.Curve> curves = barShape.curves.SelectMany(o => CurveToNative(o).Cast<DB.Curve>()).ToList();
+      if (curves is null || !curves.Any())
       {
-        appObj.Update(status: ApplicationObject.State.Failed, logItem: "Rebar host not found.");
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: "Could not convert any shape curves");
         return appObj;
       }
 
-      if (docObj != null)
+      // get the rebar plane norm from the curves
+      XYZ normal = XYZ.BasisZ;
+      if (speckleRebar.normal is not null)
       {
-        rebar = (DB.Structure.Rebar)docObj;
-        // if the number of curves don't match, we need to create a new bar
-        if (rebar.Quantity != closedCurves.Count + openCurves.Count)
-          rebar = null;
+        normal = VectorToNative(speckleRebar.normal);
       }
 
-      // update curves if we can, only available in revit 2022
-#if REVIT2022
-      if (rebar != null)
-        rebar.MoveBarInSet(0, DB.Transform.Identity);
-#endif
-
-      // create freeform from curveloops if they exist
-      if (curveLoops.Count > 0)
+      // create the rebar
+      DB.Structure.Rebar rebar = null;
+      try
       {
-        var result = RebarFreeFormValidationResult.Success;
-        rebar ??= DB.Structure.Rebar.CreateFreeForm(Doc, barType, host, curveLoops, out result);
-        if (result != RebarFreeFormValidationResult.Success)
-        {
-          appObj.Update(status: ApplicationObject.State.Failed, logItem: "Freeform Rebar could not be created from closed curves.");
-          return appObj;
-        }
+        rebar = DB.Structure.Rebar.CreateFromCurves(
+          Doc,
+          barStyle,
+          barType,
+          startHook,
+          endHook,
+          CurrentHostElement,
+          normal,
+          curves,
+          startHookOrientation,
+          endHookOrientation,
+          true,
+          true
+        );
       }
-      else if (openCurves.Count > 0)
+      catch (Autodesk.Revit.Exceptions.ArgumentNullException nullEx)
```

Comment:
I wonder if an alias might help readability here, no strong feeling but RevitException.ArgumentNullExecption might be a bit easier to read, and so on...

it's great to catch all the relevant exceptions, but it looks like we treat most the same...

This is probably a good case for catching a generic (preferably Revit) exception and then pattenring matching on the type..  or...

There's also this handy when syntax, which might be quite new and maybe not usable yet (I only just saw it)

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/when
https://code-maze.com/csharp-catch-multiple-exceptions/

This should improve the catch readability quite a bit.

Reply:
Ah, I was handling the specific exceptions when debugging to see why sometimes the `Create` method was throwing, but they ended up not being helpful. Will revert to catching a generic exception since we are handling them the same for now.